src/System.Net.HttpListener/tests/HttpListenerAuthenticationTests.cs

```diff
@@ -404,6 +424,24 @@ public void Realm_SetDisposed_ThrowsObjectDisposedException()
             return clientTask.Result;
         }
 
+        public async Task<HttpResponseMessage> AuthenticationFailureAsyncContext(HttpClient client, HttpStatusCode errorCode)
+        {
+            Task<HttpResponseMessage> clientTask = client.GetAsync(_factory.ListeningUrl);
+            Task<HttpListenerContext> serverTask = _listener.GetContextAsync();
+
+            // The client task should complete first - the server should send a 401 response.
```

Comment:
If we expect a 401, we should explicitly check for it. However, the expected status is passed into this function, so the comment is incorrect.